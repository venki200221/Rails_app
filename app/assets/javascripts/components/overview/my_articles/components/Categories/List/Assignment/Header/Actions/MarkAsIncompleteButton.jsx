import React from 'react';
import PropTypes from 'prop-types';

const update = ({
  assignment, courseSlug,
  handleUpdateAssignment, refreshAssignments
}) => async () => {
  const statuses = assignment.assignment_all_statuses;
  const prev = statuses[statuses.length - 2];

  await handleUpdateAssignment(assignment, prev);
  await refreshAssignments(courseSlug);
};

export const MarkAsIncompleteButton = props => (
  <div>
    <button
      className="button danger small"
      onClick={update(props)}
    >
      Mark as Incomplete
    </button>
  </div>
);

MarkAsIncompleteButton.propTypes = {
  // props
  assignment: PropTypes.object.isRequired,
  courseSlug: PropTypes.string.isRequired,

  // actions
  handleUpdateAssignment: PropTypes.func.isRequired,
  refreshAssignments: PropTypes.func.isRequired,
};

export default MarkAsIncompleteButton;
