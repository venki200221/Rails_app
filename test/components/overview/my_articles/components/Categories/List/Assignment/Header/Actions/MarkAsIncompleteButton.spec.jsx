import React from 'react';
import { shallow } from 'enzyme';
import '../../../../../../../../../testHelper';

import MarkAsIncompleteButton from '../../../../../../../../../../app/assets/javascripts/components/overview/my_articles/components/Categories/List/Assignment/Header/Actions/MarkAsIncompleteButton';

describe('MarkAsIncompleteButton', () => {
  const update = jest.fn();
  const props = {
    assignment: { assignment_all_statuses: [] },
    courseSlug: 'course/slug',
    handleUpdateAssignment: update,
    refreshAssignments: jest.fn()
  };
  it('should show the button', () => {
    const component = shallow(<MarkAsIncompleteButton {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should update the assignment on button click', async () => {
    const component = shallow(<MarkAsIncompleteButton {...props} />);
    const button = component.find('button');

    await button.props().onClick();
    expect(update.mock.calls.length).toEqual(1);
  });
});
