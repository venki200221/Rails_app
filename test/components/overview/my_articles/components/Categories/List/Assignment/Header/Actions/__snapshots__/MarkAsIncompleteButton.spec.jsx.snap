// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkAsIncompleteButton should show the button 1`] = `ShallowWrapper {}`;
